From c81555201880414ee77cbf09dbbbe40843e3ff1f Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Fri, 14 Nov 2008 14:10:15 -0800
Subject: Staging: comedi: set up infrastructure for individual drivers

From: Greg Kroah-Hartman <gregkh@suse.de>

This adds two "generic" Kconfig options, one for PCI and one for USB
comedi drivers, and the Makefile infrastructure as well.  This is so
we can start adding individual comedi drivers and add them to the build
properly.

Cc: Ian Abbott <abbotti@mev.co.uk>
Cc: Frank Mori Hess <fmhess@users.sourceforge.net>
Cc: David Schleef <ds@schleef.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/Kconfig          |   14 ++++++++++++++
 drivers/staging/comedi/Makefile         |    1 +
 drivers/staging/comedi/drivers/Makefile |    2 ++
 3 files changed, 17 insertions(+)

--- /dev/null
+++ b/drivers/staging/comedi/drivers/Makefile
@@ -0,0 +1,2 @@
+# Makefile for individual comedi drivers
+#
--- a/drivers/staging/comedi/Kconfig
+++ b/drivers/staging/comedi/Kconfig
@@ -11,3 +11,17 @@ config COMEDI_RT
 	default N
 	---help---
 	  Enable Real time support for the Comedi core.
+
+config COMEDI_PCI_DRIVERS
+	tristate "Comedi PCI drivers"
+	depends on COMEDI && PCI
+	default N
+	---help---
+	  Enable lots of comedi PCI drivers to be built
+
+config COMEDI_USB_DRIVERS
+	tristate "Comedi USB drivers"
+	depends on COMEDI && USB
+	default N
+	---help---
+	  Enable lots of comedi USB drivers to be built
--- a/drivers/staging/comedi/Makefile
+++ b/drivers/staging/comedi/Makefile
@@ -2,6 +2,7 @@ obj-$(CONFIG_COMEDI) += comedi.o
 obj-$(CONFIG_COMEDI_RT) += comedi_rt.o
 
 obj-$(CONFIG_COMEDI)	+= kcomedilib/
+obj-$(CONFIG_COMEDI)	+= drivers/
 
 comedi-objs :=		\
 	comedi_fops.o	\
